app/controllers/products_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ProductsController < ApplicationController
+
+	def index
+		@products = Product.order(id: :asc)
+		render :index
+	end
+
+	def show
+    @vendor = Vendor.find(params[:id])
+    render :detail_view
+  end
+
+	def new
+    @vendors = Vendor.select(:id, :name).order(:name)
+    @product = Product.new
+    render :create_product
+	end
+
+	def create
+    @product = Product.create(product_create_params[:product])
+    redirect_to products_show_path(@product.vendor_id)
+	end
+
+  def edit
+    @product = Product.find(params[:id])
+    @vendors = Vendor.new
+    render :edit
+  end
+
+  def update
+    @product = Product.find(params[:id])
+    @product.update(product_update_params[:task])
+    redirect_to products_path
+  end
+
+	def destroy
+   killed_record = Product.destroy(params[:id])
+   if params[:id] = true 
+     redirect_to products_show_path(killed_record.vendor_id)
+   end 
+	end
+
+  def product_create_params
+    params.permit(product: [:item, :vendor_id]) # these need to be finished with which params need to be let through
```

Comment:
I think these comments could be deleted, maybe they were left in from before it was done? For comments about tasks left to do if you type '# TODO' before your comment most text editors will highlight that TODO in a different color so its easy to find in your code later and also delete once done